ultraplot/constructor.py

```diff
@@ -845,65 +850,103 @@ def Cycle(*args, N=None, samples=None, name=None, **kwargs):
+            self._build_cycler(dicts + (props,))
 
-    # Get a cycler from a colormap
-    # NOTE: Passing discrete=True does not imply default_luminance=90 because
-    # someone might be trying to make qualitative colormap for use in 2D plot
-    else:
+    def _handle_colormap_args(self, args, props, kwargs, samples, name):
+        """Handle case when arguments are for creating a colormap."""
         if isinstance(args[-1], Number):
             args, samples = args[:-1], _not_none(
                 samples_positional=args[-1], samples=samples
-            )  # noqa: #501
+            )
+
+        cmap = self._create_colormap(args, name, samples, kwargs)
+        dict_ = {"color": [c if isinstance(c, str) else to_hex(c) for c in cmap.colors]}
+        self._build_cycler((dict_, props))
+        self.name = _not_none(name, cmap.name)
+
+    def _create_colormap(self, args, name, samples, kwargs):
+        """Create a colormap from the given arguments."""
         kwargs.setdefault("listmode", "discrete")
         kwargs.setdefault("filemode", "discrete")
-        kwargs["discrete"] = True  # triggers application of default 'samples'
+        kwargs["discrete"] = True
         kwargs["default_luminance"] = DEFAULT_CYCLE_LUMINANCE
-        cmap = Colormap(*args, name=name, samples=samples, **kwargs)
-        name = _not_none(name, cmap.name)
-        dict_ = {"color": [c if isinstance(c, str) else to_hex(c) for c in cmap.colors]}
-        dicts = (dict_,)
-
-    # Update the cyler property
-    dicts = dicts + (props,)
-    props = {}
-    for dict_ in dicts:
-        for key, value in dict_.items():
-            props.setdefault(key, []).extend(value)
-
-    # Build cycler with matching property lengths
-    maxlen = np.lcm.reduce([len(value) for value in props.values()])
-    props = {key: value * (maxlen // len(value)) for key, value in props.items()}
-    cycle = cycler.cycler(**props)
-    cycle.name = _not_none(name, "_no_name")
-
-    return cycle
+        return Colormap(*args, name=name, samples=samples, **kwargs)
+
+    def _is_all_cyclers(self, args):
+        """Check if all arguments are Cycler objects."""
+        return all(isinstance(arg, cycler.Cycler) for arg in args)
+
+    def _build_cycler(self, dicts):
+        """Build the final cycler from the given dictionaries."""
+        props = {}
+        for dict_ in dicts:
+            for key, value in dict_.items():
+                props.setdefault(key, []).extend(value)
+        # Build cycler with matching property lengths
+        # Ensure at least a default color property exists
+        if not props:
+            props = {"color": ["black"]}
+
+        # Build cycler with matching property lengths
+        lengths = [len(value) for value in props.values()]
+        maxlen = np.lcm.reduce(lengths)
+        props = {key: value * (maxlen // len(value)) for key, value in props.items()}
+        mcycler = cycler.cycler(**props)
+        super().__init__(mcycler)
```

Comment:
Can you add a comment on what this call does exactly? Does it use matplotlib's init function to update the current cycle to the computed one?

Reply:
`cycler.cycler` is not a class but a function in mpl. The logic of the init from the `Cycle` class is not compatible with the function -- that is we need to pass the output of cycler directly to the parent of the class we introduced.

Reply:
Cool. Let's add a comment on this in the code.